### DISCUSSION TITLE
[Feature] Not showing on "Open from" menu in Android

### DISCUSSION THREAD

**Discussion by Majestic7979** [immich-app/immich#3400] [Feature Request]
### The bug

First things first - thank you Alex for creating this project. I am literally in love with it! So great. I tried Photoprism and didn't like it one bit. Having to learn their cumbersome UI was frustrating to say the least.

You have something that all of us are used to already (GP) and an app that works mostly great. One thing missing from the app, is that it doesn't show as a source on the file picker like OneDrive, Google Drive, and Google Photos.

Is this a bug, or did you just not integrate the feature? I assume it's there but something is not working... If it's not there please definitely add it! Because this is how a user can browse for photos to upload to many apps on their device. For example for some dating apps, if you want to upload photos to your profile you will see this window, and if Immich is not a source then it's not going to be possible to upload any images hosted in Immich.

While I'm here... Another thing that would be amazing is a desktop sync app for Windows 11, so I can just point it to a folder and it will watch and keep it in sync. That would be incredible!

I am also donating weekly - something I've never done before. I think you deserve it. Keep up the good work!

![Screenshot_20230723-152134](https://github.com/immich-app/immich/assets/128995501/0ae3e488-4fd3-4f2d-8bf3-3e96fec0645f)

### The OS that Immich Server is running on

Docker on Debian 12

### Version of Immich Server

v1.68.0

### Version of Immich Mobile App

1.68.0 build .91

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Not applicable
```

### Your .env content

```Shell
Not applicable
```

### Reproduction steps

```bash
Open app on Android where you can upload photos... the Open From menu does not have Immich as a source. You can see that Nextcloud is there, along with Google Photos... Immich should be there too.
```

**Comment by alextran1502** [comment]
This is the feature we haven't implemented yet

**Comment by Majestic7979** [comment]
Is there a workaround because when I need to upload something to an all, there is no way as Immich is not a browsable source. So it's impossible to replace GP until this is done because I need to share images all the time. Some things like email it's possible to use the share sheet, same as for WhatsApp. But when you need to upload from within a third party for example to choose a new avatar for the app user profile then it's impossible without this feature. It breaks a lot of functions that users might need. Until this is implemented it cannot fully replace GP.

**Comment by alextran1502** [comment]
The workaround is to download those photos to your device and then open from your native gallery viewer

**Reply by Majestic7979** [reply]
That's a pity. It's too cumbersome to do that, involving several extra steps just for a simple upload. I'll have to keep both apps on the device until this is implemented. Kills the privacy-respecting aspect for me. I was aware that workaround could be used, I thought there might be another way. Hopefully you can prioritize it.

**Reply by alextran1502** [reply]
We all aim to get to feature parity with GPhotos eventually so eventually this will be implemented :)

**Comment by bo0tzz** [comment]
Duplicate of #1641
